cli/tools/pm/cache_deps.rs

```diff
@@ -94,7 +100,16 @@ pub async fn cache_top_level_deps(
             });
           }
         }
-        "npm" => roots.push(specifier.clone()),
+        "npm" => {
+          if workspace_npm_package_names.contains(
+            specifier.as_str()[4..]
+              .trim_start_matches('/')
+              .trim_end_matches('/'),
```

Comment:
I think this won't work if someone specifies a subpath? Probably we should parse to `NpmPackageReqReference` here.